### ISSUE TITLE
Feature Request: Support Lambert W x F distributions

### ISSUE THREAD

**Issue by gmgeorg** [StatMixedML/XGBoostLSS#54]
It would be great if XGBoostLSS can support Lambert W x F distributions; particularly useful are Lambert W x Gaussian distributions (Tukey's h is a special case of this for $\alpha=1$ and $h = \delta$) as they can be used to transform data to normally distributed data, even if original data is (very) heavy tailed.

In XGBoostLSS context I can see this being useful any time where normal regression might be too restrictive to give correct tail probability estimates (e.g., low sample size; financial data) and one can inspect $\delta$ predictions from XGBoostLSS for which parts of space have long/heavy tail (more uncertainty) than others.  Secondly, skewed/heavy-tail Lambert W x gamma distributions are useful to impose heavier right tail for survival like problems.

I'm not aware of a `pytorch` implementation of Lambert W function, let alone Lambert W x F distributions.  TensorFlow has both implemented; [`scipy.special.lambertw`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.lambertw.html) implements the Lambert W function.

If a `pytorch` implementation of the `distribution` is required to make this work in XGBoostLSS, then as an alternative AFAICT this should be possible to accomplish using normalizing flows, with the heavy-tail Lambert W transformation as a specific normalizing flow function. 

**References**

* [heavy-tail Lambert W x F distributions](https://www.hindawi.com/journals/tswj/2015/909231/), Goerg (2015)

* *LambertW* R package: https://github.com/gmgeorg/LambertW

* TensorFlow probability implementation of [Lambert W bijectors](https://www.tensorflow.org/probability/api_docs/python/tfp/bijectors/LambertWTail) and [Lambert W x Gaussian `Distribution`](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/LambertWDistribution)

* gaussianization layers based on Lambert W x F transformations/distributions: https://openreview.net/forum?id=OXP9Ns0gnIq
